### PR TITLE
feat(pmp-web): add repository data-access #149

### PR DESCRIPTION
**Pull request by Stefaneek**

### CODE REVIEW

File: libs/pmp-web/repository/domain/src/lib/interfaces/repository.model.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface Repository {
+  id: string;
+  fullName: string;
+  name: string;
+  owner: string;
+  pictureUrl: string;
+  maxLines?: number;
+  maxWaitingTime?: number;
+  // prs: PrEntity[] = [];
```
Comment (Sikora00) [inline libs/pmp-web/repository/domain/src/lib/interfaces/repository.model.ts:9]: I think we can remove that line

File: libs/pmp-web/repository/domain/src/lib/payloads/get-repository-collection.payload.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// tslint:disable-next-line:no-empty-interface
```
Comment (Sikora00) [inline libs/pmp-web/repository/domain/src/lib/payloads/get-repository-collection.payload.ts:1]: maybe just remove this rule from ts lint

File: libs/pmp-web/repository/data-access/src/lib/+state/repository.effects.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    onError: (action: fromRepositoryActions.GetRepositoryCollection, error: HttpErrorResponse) => {
+      return new fromRepositoryActions.GetRepositoryCollectionFail(error);
+    }
+  });
+  @Effect()
+  addRepository$ = this.dp.fetch(fromRepositoryActions.Types.AddRepository, {
+    run: (action: fromRepositoryActions.AddRepository) => {
+      return this.repositoryDataService
+        .addRepository(action.payload)
+        .pipe(map(() => new fromRepositoryActions.AddRepositorySuccess()));
```
Comment (Sikora00) [inline libs/pmp-web/repository/data-access/src/lib/+state/repository.effects.ts:28]: some reload on success?

File: libs/pmp-web/repository/domain/src/lib/interfaces/repository.model.ts (hunk 1 above)
Comment (Stefaneek) [inline libs/pmp-web/repository/domain/src/lib/interfaces/repository.model.ts:9]: Need to discuss it with @Sikora00

**Pull request merged**